### PR TITLE
Use NPM registry for querying all packages

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: bayesian_jobs/handlers/npm_popular_analyses.py [hunk 1]
```diff
@@ -6,37 +7,72 @@
...
-        _count = sorted(map(int, _count.split("-")))
-        if len(_count) == 1:
-            _min = 0
-            _max = _count[0]
-        elif len(_count) == 2:
-            _min = _count[0] - 1
-            _max = _count[1]
-        else:
-            raise ValueError("Bad count %r" % count)
+        # this solution might be ugly, but is quiet efficient compared to downloading info
```
Comment (msrb) [inline bayesian_jobs/handlers/npm_popular_analyses.py]: Note CoachDB supports `skip` and `limit` query parameters:
http://docs.couchdb.org/en/2.0.0/api/database/bulk-api.html

I believe those could be used to skip all records up to `start` and limit the number of results returned.

Using those query parameters could simplify the code and definitely make it more efficient.

File: bayesian_jobs/handlers/npm_popular_analyses.py [hunk 2]
```diff
@@ -50,3 +86,27 @@ def execute(self, popular=True, count=None, nversions=None, force=False):
...
+        :param nversions: how many versions of each project to schedule
+        :param force: force analyses scheduling
+        """
+        _count = count or str(self._DEFAULT_COUNT)
+        _count = sorted(map(int, _count.split("-")))
+        if len(_count) == 1:
+            _min = 0
+            _max = _count[0]
+        elif len(_count) == 2:
+            _min = _count[0] - 1
```
Comment (msrb) [inline bayesian_jobs/handlers/npm_popular_analyses.py]: I must have missed something, but why `-1`?

File: bayesian_jobs/handlers/npm_popular_analyses.py (hunk 1 above)
Comment (msrb) [inline bayesian_jobs/handlers/npm_popular_analyses.py]: I am OK with optimizing this later. I just wanted to point out that there is a way how to make this more efficient.

**Comment by msrb** [comment]
No real blockers, just few comments.

File: bayesian_jobs/handlers/npm_popular_analyses.py (hunk 2 above)
Comment (fridex) [inline bayesian_jobs/handlers/npm_popular_analyses.py]: Haven't changed this, just moved (basically this is indexing from 1 on API).

File: bayesian_jobs/handlers/npm_popular_analyses.py (hunk 1 above)
Comment (fridex) [inline bayesian_jobs/handlers/npm_popular_analyses.py]: I somehow missed those parameters when I was working on this. I don't like "later", so I will rework it. Thanks.

**Comment by fridex** [comment]
@msrb updated with limit and skip, tested

**Pull request merged**